vms/platformvm/state/state.go

```diff
@@ -1964,21 +2115,6 @@ func (s *state) writeCurrentStakers(updateValidators bool, height uint64) error
 				staker := validatorDiff.validator
 				weightDiff.Amount = staker.Weight
 
-				// Invariant: Only the Primary Network contains non-nil public
-				// keys.
-				if staker.PublicKey != nil {
-					// Record that the public key for the validator is being
-					// added. This means the prior value for the public key was
-					// nil.
-					err := s.flatValidatorPublicKeyDiffsDB.Put(
```

Comment:
In the existing code, we only write to `flatValidatorPublicKeyDiffsDB` if `staker.PublicKey != nil`. In the new code,